packages/devkit/src/utils/replace-package.ts

```diff
@@ -0,0 +1,166 @@
+import type { Tree } from 'nx/src/generators/tree';
+import type { PackageJson } from 'nx/src/utils/package-json';
+import { requireNx } from '../../nx';
+import { NX_VERSION } from './package-json';
+import { visitNotIgnoredFiles } from '../generators/visit-not-ignored-files';
+import { basename } from 'path';
+
+const {
+  getProjects,
+  updateProjectConfiguration,
+  readNxJson,
+  updateNxJson,
+  updateJson,
+} = requireNx();
+
+export function replaceNrwlPackageWithNxPackage(
+  tree: Tree,
+  oldPackageName: string,
+  newPackageName: string
+): void {
+  replacePackageInDependencies(tree, oldPackageName, newPackageName);
+
+  replacePackageInProjectConfigurations(tree, oldPackageName, newPackageName);
+
+  replacePackageInNxJson(tree, oldPackageName, newPackageName);
+
+  replaceMentions(tree, oldPackageName, newPackageName);
+}
+
+function replacePackageInDependencies(
+  tree: Tree,
+  oldPackageName: string,
+  newPackageName: string
+) {
+  visitNotIgnoredFiles(tree, '.', (path) => {
+    if (basename(path) !== 'package.json') {
+      return;
+    }
+
+    updateJson<PackageJson>(tree, path, (packageJson) => {
+      for (const deps of [
+        packageJson.dependencies ?? {},
+        packageJson.devDependencies ?? {},
+        packageJson.peerDependencies ?? {},
+        packageJson.optionalDependencies ?? {},
+      ]) {
+        if (oldPackageName in deps) {
+          deps[newPackageName] = NX_VERSION;
+          delete deps[oldPackageName];
+        }
+      }
+      return packageJson;
+    });
```

Comment:
Shouldn't this be similarly to others be marked with `needsUpdate` and if `true` run `installPackagesTask` at the end?

Or is this only meant to be run within the migration (which prompts for `install`)?

Reply:
It's only used for migrations for now... I could make it return the task though.